style: complete profile page layout

sounmind opened the pull request:
## Summary

This is style changes for profile page layout based on this [Figma design](https://www.figma.com/design/62ZRtfpGdAz2uuWwVMAmQg/VMS-Project?node-id=219-5922&t=XwaPeRQtdE0lmi9v-0)

## Tests

We've coded with @madcampos, it's approved...
We did not test mobile version yet.

## Additional information

sounmind committed:
style: complete and extract profile page layout

Co-authored-by: ImmaCheetah <ImmaCheetah@users.noreply.github.com>
Co-authored-by: Marco Campos <me@madcampos.dev>
Co-authored-by: Evan Suhyeong Lee <devleesche@gmail.com>

kbventures commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,241 @@
+			</details>
+		</div>
+
+		<Button isPrimary id='submit-button' type='submit'>
```
@sounmind @ImmaCheetah 

The button width isn't matchign Figma design on the mobile and full screen version:

![image](https://github.com/user-attachments/assets/bb418673-8937-410c-aedb-1e94215c3000)

See the Figma design:

![image](https://github.com/user-attachments/assets/384dc40d-2d45-4dd6-aebb-9cbe01ca8bb7)

The left and right spacing is not matching the mobile FIgma design.

kbventures commented:
@ImmaCheetah @madcampos 

Using padding instead of margin between elements can cause accessibility issues, especially for users with high zoom levels or screen magnifiers. This is because padding increases the clickable area inside an element, which can lead to overlapping interactive regions. Margin is better for spacing between elements to maintain clear separation.

ImmaCheetah committed:
wib: add grid to in a nutshell layout

Co-authored-by: Manvinderjit Rupal <manvinderjit@users.noreply.github.com>
Co-authored-by: Will Frankland <will-frankland@users.noreply.github.com>

manvinderjit committed:
WIP: Add CSS styles for inputs of In a nutshell

Co-authored-by: Evan Suhyeong Lee <sounmind@users.noreply.github.com>
Co-authored-by: ImmaCheetah <ImmaCheetah@users.noreply.github.com>

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -56,23 +56,57 @@
+					height: 2.33rem;
+					box-sizing: border-box;
+					-webkit-box-sizing:border-box;
+					-moz-box-sizing: border-box;
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** Please remove vendor prefixed properties. There is only a handful of properties that still need vendor prefixes, and `box-sizing` is not one of them.

It has been widely available on browsers since July 2015‏.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -56,23 +56,57 @@
+						height: 1rem;
+						width: 1rem;
+					}
+				} */
```
Unfortunately there is no cross browser way of styling the small arrow added by the datalist element on chromium browsers. Just leave it as is. Not the prettiest, but it works 🤷‍♂️

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -56,23 +56,57 @@
 
-			.input-row-required {
+			.input-required {
+				& > span {
```
```suggestion
			.input-required span {
```

manvinderjit committed:
WIP: Style checkbox to toggle slider

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -119,7 +128,61 @@
 				display: grid;
 				gap: var(--spacing-xs);
-				grid-template-columns: 1fr 8fr;
+				grid-template-columns: repeat(1fr, 2fr);
```
I think there is a typo here... The `repeat` will be ignored.

madcampos replied:
There is still a typo here...

```suggestion
				grid-template-columns: repeat(2, 1fr);
```

See: https://developer.mozilla.org/en-US/docs/Web/CSS/repeat

manvinderjit committed:
WIP: Add Photo Preview in Avatar Section

manvinderjit committed:
fix: Fix padding, body margin

Fix padding and remove body margins of the Complete Profile page.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,256 @@
+			/* Media query for smaller screens */
+			@media (max-width: 768px) { .details-content-grid { grid-template-columns: 1fr; } }
+
+			.input-required {
+				& > span {
+					color: var(--color-accent);
+					font-size: 0.56rem;
+					font-weight: 400;
+					height: 1rem;
+					letter-spacing: 0;
+					line-height: 160%;
+				}
+				align-items: center;
+				display: flex;
+				justify-content: space-between;
```
```suggestion
			.input-required {
				align-items: center;
				display: flex;
				justify-content: space-between;
				
				& > span {
					color: var(--color-accent);
					font-size: 0.56rem;
					font-weight: 400;
					height: 1rem;
					letter-spacing: 0;
					line-height: 160%;
				}
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** Please put the element styles before any nesting.‏

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,256 @@
+					width: 100%;
+				}
+
+				/* .text-input#pronouns {
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** Please let's keep the code tidy and remove unused and commented out pieces.‏

madcampos commented on src/pages/complete-profile/style.css:
```diff
@@ -0,0 +1 @@
+body { margin: 0; }
```
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** Let's centralize "CSS resets", please refer to: https://github.com/torontojs/communityhub-fe/pull/33‏

manvinderjit replied:
Done!

manvinderjit committed:
style: add styling to More Information section

Co-authored-by: Evan Suhyeong Lee <sounmind@users.noreply.github.com>

manvinderjit committed:
feat: add basic skills function to Complete Profile Page

manvinderjit committed:
style: update CompleteProfile styles

- Add hover styles to skills.
- Fix slider toggle button width for uniform padding.
- Allow skills to be added when spacebar is pressed.

manvinderjit committed:
style: update Social Accounts styles

sounmind committed:
fix: prevent submit error

Co-authored-by: Manvinderjit Rupal <manvinderjit@users.noreply.github.com>

sounmind committed:
fix: resolve merge conflicts

sounmind committed:
fix: key prop warnings

manvinderjit commented:
@madcampos @kbventures Please re-review this! Thanks!

manvinderjit committed:
fix: remove unwanted and commented styles

manvinderjit committed:
fix: give skills unique IDs

madcampos commented:
Hey folks, I ran the code locally and used NVDA to test it out...

Here is how it is read by a screen reader:

https://github.com/user-attachments/assets/3a911e15-9119-4199-b9a7-8aca2fe57a33

Notice some things:

1. The upload avatar button is entirely skipped, it is not accessible through keyboard navigation
2. All of the social media buttons announce as "button", it doesn't say what they do
3. The fields marked as required have the "required" part read twice

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				margin-right: 0.4rem;
+				min-width: 36px;
+				transition: 300ms ease-in-out;
+				width: 36px;
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** Please move the pixel values to use rem.‏

manvinderjit replied:
Done!

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+
+			.slider-checkbox:checked + label::after {
+				background-color: white;
+				left: 20px;
```
Same here

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+			max-height: 128px;
+			max-width: 128px;
+			min-height: 52px;
+			min-width: 52px;
```
And here

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				aspect-ratio: 1;
+				border-radius: 0.5rem;
+				height: 100%;
+				object-fit: cover;
```
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** we should use `contain` instead of `cover` here as it does not crop images that are not square. `cover` will make the image fill all the available space but crop it to fit the space, where `contain` will resize the image to fit inside the area.

On the screenshot the image is rectangular, where the preview is square and crops the image:
![image](https://github.com/user-attachments/assets/d90faddf-638c-4601-8697-033e7cad5fe4)
‏

manvinderjit replied:
@madcampos The reason for using cover was so that the border radius applies to the image. Because if the image doesn't have a 1:1 aspect ratio, the border radius effect is virtually invisible with contain. See attached image for details.
![Screenshot from 2025-05-06 15-06-07](https://github.com/user-attachments/assets/988bdd83-b511-43ef-879d-0f4b569aaabe)
![Screenshot from 2025-05-06 15-06-15](https://github.com/user-attachments/assets/38ae8b09-cd23-4bb2-8162-7517e21d2e11)
![Screenshot from 2025-05-06 15-06-25](https://github.com/user-attachments/assets/ad1d5d76-d506-4c34-9725-d5d0932a4f04)

manvinderjit replied:
Do you recommend making the width fluid and height fixed with `contain`?

madcampos replied:
Use this: https://github.com/madcampos/sdrlog/blob/main/src/components/SdrCard/style.css#L132-L152

Wrap the `img` in a `picture`, then the `picture` sets the size you want, the `img` can grow up to it, but is bound to the box size. It will keep the aspect ratio though.

manvinderjit replied:
Got it, thanks.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+			min-width: 244px;
+			width: 244px;
+
+			& input[type='file'] { display: none; }
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** This removes the input from the DOM, making it not accessible for keyboard users. A solution is do something similar to the checkbox, where it is visibly hidden, but not removed from the DOM.‏

manvinderjit replied:
@madcampos  Change the logic and visibility, should resolve now.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				grid-template-columns: repeat(1fr, 2fr);
+
+				& label {
+					font-size: var(--text-sm);
+					letter-spacing: 0;
+					line-height: 160%;
+				}
+			}
+
+			.slider-checkbox {
+				opacity: 0;
+				position: absolute;
+			}
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** The checkbox doesn't have a focus state. So when keyboard users navigate to it, it does not show any indicator that the element is selected.‏

Here is a video recording of it:

https://github.com/user-attachments/assets/ff75584d-6ec0-4823-ae9c-10f0098e00b9

manvinderjit replied:
Implemented the change.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				font-size: var(--text-xs);
+				font-weight: var(--weight-semibold);
+				letter-spacing: 0;
+				line-height: 160%;
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** messing with `line-height` can cause issue due to how it is calculated on CSS. In regular image/text editing software it is calculated based on the x-height, meaning, the height of lowercase characters, not counting ascenders, nor descenders (like the stick going up on "d" or down on "p").

On CSS the line height is calculated by drawing the whole text inside a box, and then adding equal space on top and bottom from it. So ascenders and descenders count for CSS line height, where for other tools they do not.

So, it is not a very good idea to change the line height as it may cause more troubles than it is worth.

manvinderjit replied:
Removed most cases of line-height.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				background-color: var(--color-background);
+				border: var(--border);
+				border-color: var(--color-black);
+			}
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** The "remove photo" button is rendering very differently than the label with the same style. Maybe setting the button `appearance` and `font` properties may help.‏

manvinderjit replied:
@madcampos  Replaced label with button, should resolve now.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+		grid-template-columns: repeat(3, 1fr);
+		justify-content: space-between;
+
+		label[for='linkedInProfile'], label[for='githubProfile'], label[for='portfolio'] {
```
```suggestion
		label:is([for='linkedInProfile'], [for='githubProfile'], [for='portfolio']) {
```

The `:is` selector works as a grouping selector.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+			flex-direction: column;
+			grid-column: 1 / span 3;
+
+			> label {
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->

```suggestion
			& > label {
```

**nitpick:** To disambiguate any complex selector that is not the children selector (just a space between selectors), it is better to add the `&` here.‏

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				padding: var(--spacing-xs);
+				resize: none;
+
+				> span {
```
Same here

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+					padding-right: var(--spacing-xs);
+					transition: all ease-in-out 300ms;
+
+					> button {
```
And here

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+					align-items: center;
+					background-color: var(--color-accent);
+					border: solid var(--border-base) var(--color-accent);
+					border-radius: 99px;
```
```suggestion
					border-radius: 100vmax;
```

Neat little trick to make things round 😉

It uses the largest viewport unit (`vmax`) and makes it as big as the whole viewport (100 of the unit).

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,401 @@
+				input#skill {
+					background: none;
+					border: solid var(--border-base) var(--color-grey);
+					border-radius: 99px;
```
```suggestion
```

Don't add `border-radius` here as it crops the input text.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,399 @@
+								<div>
+									<label htmlFor='email' className='input-required'>
+										E-mail
+										<span>REQUIRED</span>
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** This "required" text gets read twice by screen readers, first from the input being marked as required, second from the `span` here. We can change it to be added with CSS pseudo elements (`::before` or `::after`).‏

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,399 @@
+									<div className='dob-wrapper'>
+										<label htmlFor='month'>Month</label>
+										<select id='month' name='month'>
+											<option value='01'>January</option>
```
```suggestion
											<option selected disabled hidden value=''>Select a month</option>
											<option value='01'>January</option>
```

This is a neat trick to add an option that doesn't show up but still adds a "placeholder" for the user.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,399 @@
+										</select>
+										<label htmlFor='day'>Day</label>
+										<select id='day' name='day'>
+											<option value='01'>1</option>
```
```suggestion
											<option selected disabled hidden value=''>Select a day</option>
											<option value='01'>1</option>
```

Same trick here.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,399 @@
+								</div>
+								<div className='slider-wrapper'>
+									<div className='slider-checkbox-row'>
+										<input
+											id='isBasedOnGTA'
+											name='isBasedOnGTA'
+											type='checkbox'
+											className='slider-checkbox'
+										/>
+										<label htmlFor='isBasedOnGTA'>
+											I'm based in Toronto or Greater Toronto Area
```
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** Here an option is to put the checkbox inside the label along with a span containing the text for the checkbox.‏

manvinderjit replied:
They are in this order because I am adding the toggle sliders using CSS pseudo elements. The input checkbox is invisible here anyways.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,399 @@
+					{ label: 'Complete your profile' }
+				]}
+			/>
+			<form action='' encType='multipart/form-data' id='complete-profile-form'>
```
```suggestion
			<form action='' encType='multipart/form-data' id='complete-profile-form' method='post'>
```

manvinderjit replied:
Will add this. Just for future reference, we are using a separate branch and PR for implementing the form submission functionality in Complete Profile. 
https://github.com/torontojs/communityhub-fe/pull/54

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,399 @@
+										id='slack-handle'
+										name='slack-handle'
+										type='text'
+										aria-description='Your slack handle to TorontoJS'
```
```suggestion
```

This is being announced twice.

manvinderjit replied:
@madcampos  Replaced aria-description with aria-label, should only read it once, please verify.

manvinderjit replied:
Also, is there an Ubuntu alternative for NVDA?

madcampos replied:
@manvinderjit yes, here is a list of options for screen readers:
- For linux there is [Orca](https://orca.gnome.org/).
- Android comes with [TalkBack](https://support.google.com/accessibility/android/answer/6007100?hl%3Den), you may need to download the app from the playstore though.
- Mac has [VoiceOver](https://support.apple.com/en-ca/guide/voiceover/welcome/mac) builtin.
- iOS also have [VoiceOver](https://support.apple.com/en-ca/guide/iphone/iph3e2e415f/ios) builtin.
- Windows has [Narrator](https://support.microsoft.com/en-us/windows/complete-guide-to-narrator-e4397a0d-ef4f-b386-d8ae-c172f109bdb1) builtin.
- There is also [JAWS](https://www.freedomscientific.com/products/software/jaws/) for windows, but it is paid.

madcampos replied:
It still gets announced twice. The browser will read whatever text is in the placeholder (I didn't knew about that, before browsers didn't read it).

Let's remove the `aria-label`

manvinderjit replied:
Yeah, it is a bit of mixed bag. I have read in multiple forums that `aria-label` supersedes the `placeholder` with only the former being announced but seems like it's not a universal thing.

madcampos replied:
It used to be the case that placeholders were not read. But maybe it is the combo Edge + NVDA

manvinderjit replied:
@madcampos Removed the `aria-label`s

manvinderjit committed:
fix: Convert pixels to rem units

manvinderjit committed:
fix: implement recommended css style changes

add & to non- children selectors in css
remove border radius on skill input
change border radius unit used on span for skills
change grouping selector label

manvinderjit committed:
fix: fix HTML elements and styles

Add placeholder values for select inputs.
Fix width of select inputs to show placeholder values.
Add span tags for required input labels using css pseudo elements.
Fix toggle slider not being highlighted on focus.
Add spacing between labels and inputs.
Add aria label to social media buttons for screen readers.
Fix slack handle being read twice by screen readers.
Fix img tag to align with object-fit: contain.
Switch Upload Photo from label to button to prevent focus skipping.
Remove line-height css property where not needed.

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,413 @@
+				left: 1.313rem;
+			}
+
+			.slider-checkbox:focus + label::before, .slider-checkbox:checked:focus + label::before { outline: auto Highlight; }
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** [`outline`](https://developer.mozilla.org/en-US/docs/Web/CSS/outline) has a similar syntax to border‏.

Please either use the shorthand syntax specifying the type, thickness, and color. Or use the longhand syntax for changing only some of the properties.

Also: `Highlight` is a [system color](https://developer.mozilla.org/en-US/docs/Web/CSS/system-color), I think it is not the best option to use here.

The two common scenarios for using system colors are:
- When using the `forced-colors` media query
- When trying to mimic an operating system design, like trying to make the page look like a Windows or MacOS application.

madcampos replied:
The focus style for the checkbox doesn't provide enough contrast and the checkbox is missalligned with the actual toggle.

<img width="101" alt="image" src="https://github.com/user-attachments/assets/bdb9b1f7-7286-4b8a-988f-c1849c22ccc1" />

Please position the input so it overlaps the visible toggle.

For the outline style, setting [`outline-offset`](https://developer.mozilla.org/en-US/docs/Web/CSS/outline-offset) helps here as it will move the outline outside of the toggle button, creating some space between the two thus increasing the contrast.

Outlines are drawn on top of the box and don't add to the box sizing, like border does, so adding an offset to the outline won't cause a [layout shift](https://developer.mozilla.org/en-US/docs/Glossary/CLS).

manvinderjit replied:
@madcampos Applied these changes.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,419 @@
+						</div>
+					</details>
+
+					{/* TODO: Need to confirm adding green check icon on successful section completion on mobile view */}
```
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** Was this confirmed?‏

manvinderjit replied:
Not yet, will bring it up with Cadu and Mariana again.

manvinderjit replied:
No update yet from design team, will implement this if and when confirmed.

madcampos replied:
Let's leave it as an "outstanging item" as it is a non blocker and we can address later

madcampos commented on src/components/CompleteProfile/CompleteProfile.css:
```diff
@@ -0,0 +1,413 @@
+
+			.input-required::after {
+				color: var(--color-accent);
+				content: 'REQUIRED';
```
```suggestion
				content: 'Required' / '';
				text-transform: uppercase;
```

The second part of the `content` property is text added to accessibility technologies for generated content. Without it all modern browsers will announce the "required" text. 🥲

It is only recently supported however (since 2024), so for other projects add it with and without the alt text.

The `text-trasnform` property does auto transformation for us.

manvinderjit replied:
I didn't know this one. Thanks for sharing.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,419 @@
+											<label htmlFor={socialInput}>
+												{socialInput}
+												<button
+													type='button'
```
```suggestion
													type='button'
																										aria-label={`Close ${socialInput} input`}
```

These button gets announced as only "button"

manvinderjit replied:
I need to start using a screen reader for catching these.

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,419 @@
+								)}
+
+								<div className='details-file-upload-buttons-wrapper'>
+									{photoFile && (
+										<span className='file-upload-success'>
+											Avatar uploaded successfully
+										</span>
+									)}
```
```suggestion
									<span ref={avatarUploadStatusRef} aria-live='polite' aria-atomic='true' role='status' className='file-upload-success'>
										{photoFile ? 'Avatar uploaded successfully' : ''}
									</span>
```

This will make the upload be announced (or should make it so). And of course, we need to add a ref here because react is a piece of dumpster fire

madcampos commented on src/components/CompleteProfile/CompleteProfile.tsx:
```diff
@@ -0,0 +1,419 @@
+const CompleteProfile = () => {
+	const fileInputRef = useRef<HTMLInputElement>(null);
+	const uploadPhotoButtonRef = useRef<HTMLButtonElement>(null);
+	const skillInputRef = useRef<HTMLInputElement>(null);
```
```suggestion
	const skillInputRef = useRef<HTMLInputElement>(null);
	const avatarUploadStatusRef = useRef<HTMLSpanElement>(null);
```

The other ref

manvinderjit committed:
fix: allow screenreaders to announce avatar upload success message

Co-authored-by: Marco Campos <me@madcampos.dev>

manvinderjit committed:
fix: apply review suggestions

Remove aria-label from slack handle to prevent double announcement by screen readers.
Add aria-label for the close button of social inputs.
Remove non-required grid around slider rows.
Fix focus outline of slider buttons.
Align checkboxes with respective slider buttons.
Fix maximum width of the wrapper.
Adjust font-weight and font-size of labels.
Replace span and p tags with labels for labelling input fields.
Fix minimum width to accommodate 320px and above screens.

manvinderjit commented:
@madcampos Hopefully should be resolved now. Let me know if something still needs work.

madcampos committed:
fix: label announcing close buttons
